libraries/core-react/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { useState, useCallback, forwardRef } from 'react'
+import DatePicker, {
+  ReactDatePickerProps,
+  registerLocale,
+} from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import enGb from 'date-fns/locale/en-GB'
+import styled from 'styled-components'
+import { Icon } from '../Icon'
+import { calendar } from '@equinor/eds-icons'
+import { PopupHeader } from './PopupHeader'
+import { datePicker as tokens, DatePickerToken } from './DatePicker.tokens'
+
+registerLocale('en-gb', enGb)
+
+export interface DatePickerProps {
+  id: string
+  value: Date | undefined | null
+  label: string
+  onChanged?: (date: Date | null) => void
+  disableFuture?: boolean
+  disableBeforeDate?: Date
+  className?: string
+  dateFormat?: string
+  placeholder?: string
+  readOnly?: boolean
+  popperPlacement?:
+    | 'auto-start'
+    | 'auto'
+    | 'auto-end'
+    | 'top-start'
+    | 'top'
+    | 'top-end'
+    | 'right-start'
+    | 'right'
+    | 'right-end'
+    | 'bottom-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'left-end'
+    | 'left'
+    | 'left-start'
+  locale?: string
+}
+
+export interface DatePickerRefProps extends ReactDatePickerProps {
+  setBlur: () => void
+  setFocus: () => void
+  setOpen: (open: boolean, skipSetBlur?: boolean) => void
+  isCalendarOpen: () => boolean
+}
+
+const ReactDatePicker = forwardRef<DatePickerRefProps, DatePickerProps>(
```

Comment:
First parameter to `forwardRef` is the type of ref, which should be `HTMLInputElement` (as `Datepicker` component is an input). Either you extend `DatePickerRefProps` with `HTMLInputElement` or use `HTMLInputElement` here.